### PR TITLE
NVSHAS-9525: Resolve existing Go linter issues in the NeuVector repository

### PR DESCRIPTION
**Pull request by williamlin-suse**
neuvector/share - part 2

### CODE REVIEW

File: share/cluster/grpc.go [hunk 1]
```diff
@@ -94,6 +94,8 @@ func NewGRPCServerTCP(endpoint string) (*GRPCServer, error) {
 		return nil, fmt.Errorf("failed to create new server credentials: %w", err)
 	}
 
+	//to avoid go-linter "SA1019: grpc.NewGZIPDecompressor is deprecated: use package encoding/gzip. (staticcheck)"
```
Comment (yasker) [inline share/cluster/grpc.go:97]: Can we add `TODO: addressing go-linter....` here? Same below, so we can address it later and not forget about it.

**Review by yasker — CHANGES_REQUESTED** [review]

File: share/cluster/grpc.go (hunk 1 above)
Comment (yasker) [reply share/cluster/grpc.go:97]: I meant...
`TODO: addressing go-linter "SA1019: grpc.NewGZIPDecompressor is deprecated: use package encoding/gzip. (staticcheck)"`

**Review by yasker — CHANGES_REQUESTED** [review]
Can you also use `git rebase` instead of merge for the new commits? There shouldn't be any `merge` commits in the PR.

**Review by jeffhuang4704 — APPROVED** [review]
looks good to me.

**Review by yasker — APPROVED** [review]

**Pull request merged**